mmseg/models/backbones/deit.py

```diff
@@ -0,0 +1,314 @@
+import math
+import warnings
+
+import torch
+import torch.nn as nn
+from mmcv.cnn import (build_norm_layer, constant_init, kaiming_init,
+                      normal_init, trunc_normal_init)
+from mmcv.cnn.bricks.transformer import FFN, MultiheadAttention
+from mmcv.runner import BaseModule, ModuleList, _load_checkpoint
+from torch.nn.modules.batchnorm import _BatchNorm
+from torch.nn.modules.utils import _pair as to_2tuple
+
+from mmseg.ops import resize
+from mmseg.utils import get_root_logger
+from ..builder import BACKBONES
+from ..utils import PatchEmbed, vit_convert
+
+from mmseg.models.backbones.vit import TransformerEncoderLayer, VisionTransformer
+
+@BACKBONES.register_module()
+class Deit(VisionTransformer):
+    """Vision Transformer variant. (Deit)
+
+    A PyTorch implement of : `DeiT: Data-efficient Image Transformers` -
+        https://arxiv.org/abs/2012.12877
+
+    Compared with ViT:
+        a) add self.dist_token attribute beside cls_token
+        b) init method of self.pos_embed: torch.zeros->torch.randn

```

Comment:
Since ViT and DeiT have plenty of the same codes, it's better to set a parameter to control whether the backbone should contain `self.dist_token`.

Reply:
Ok, I agree. I will modify vit.py accordingly.